src/core.c/Lock.pm6

```diff
@@ -9,7 +9,9 @@ my class Lock {
         method new() {
             X::Lock::ConditionVariable::New.new.throw
         }
-        method wait() { nqp::condwait(self) }
+        proto method wait(|) {*}
+        multi method wait() { nqp::condwait(self) }
+        multi method wait(&predicate) { nqp::condwait(self) while !predicate; }
```

Comment:
Why not `until predicate`?

Reply:
Because I only thought of that after I wrote the code?